tensorstore/kvstore/s3/s3_credential_provider.h

```diff
@@ -0,0 +1,112 @@
+// Copyright 2020 The TensorStore Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TENSORSTORE_KVSTORE_S3_S3_CREDENTIAL_PROVIDER_H_
+#define TENSORSTORE_KVSTORE_S3_S3_CREDENTIAL_PROVIDER_H
+
+#include <functional>
+#include <string>
+#include <string_view>
+#include <map>
+#include <memory>
+
+#include "absl/synchronization/mutex.h"
+#include "absl/time/time.h"
+#include "tensorstore/internal/http/curl_handle.h"
+#include "tensorstore/internal/http/curl_transport.h"
+#include "tensorstore/internal/http/http_transport.h"
+#include "tensorstore/util/result.h"
+
+using ::tensorstore::Result;
+
+namespace tensorstore {
+namespace internal_auth_s3 {
+
+
+class S3Credentials {
```

Comment:
Could be a struct?

Reply:
Yes, this has been changed to a struct (and passed as an arg in `S3RequestBuilder::BuildRequest`)